### PR TITLE
test: add e2e tests with kind cluster

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

- Adds e2e tests with kind cluster

**Issue Fixed**:

**Notes for Reviewers**:

### CODE REVIEW

File: scripts/ci-e2e-kind.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+create_kind_cluster () {
+  # create a cluster with the local registry enabled in containerd
+  # add encryption config and the kms static pod manifest with custom image
+  cat <<EOF | kind create cluster --retain --image kindest/node:"${KUBERNETES_VERSION}" --name "${KIND_CLUSTER_NAME}" --wait 2m --config=-
+  kind: Cluster
+  apiVersion: kind.x-k8s.io/v1alpha4
+  containerdConfigPatches:
+  - |-
+    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
```
Comment (ritazh) [inline scripts/ci-e2e-kind.sh:15]: Is this a requirement?

Comment (aramase) [reply scripts/ci-e2e-kind.sh:15]: yeah, this is required to enable local registry: https://kind.sigs.k8s.io/docs/user/local-registry/

Comment (ritazh) [reply scripts/ci-e2e-kind.sh:15]: nvm

File: tests/e2e/azure.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "cloud": "AzurePublicCloud",
+    "tenantId": "$TENANT_ID",
+    "aadClientId": "$CLIENT_ID",
+    "aadClientSecret": "$CLIENT_SECRET"
+}
```
Comment (ritazh) [inline tests/e2e/azure.json:6]: nit

File: tests/e2e/kms.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          cpu: 100m
+          memory: 128Mi
+        limits:
+          cpu: 4
+          memory: 2Gi
+      volumeMounts:
+        - name: etc-kubernetes
+          mountPath: /etc/kubernetes
+        - name: etc-ssl
+          mountPath: /etc/ssl
```
Comment (ritazh) [inline tests/e2e/kms.yaml:32]: where do we use this?

**Review by ritazh — APPROVED** [review]
Just few nits. but otherwise LGTM

File: tests/e2e/kms.yaml (hunk 3 above)
Comment (aramase) [reply tests/e2e/kms.yaml:32]: This is required to access root CA certs. Without this volume kv calls will fail with:
```bash
2021-02-09T01:52:20.524238Z stderr F E0209 01:52:20.523812       1 server.go:50] "failed to encrypt" err="failed to encrypt, error: keyvault.BaseClient#Encrypt: Failure sending request: StatusCode=0 -- Original Error: context deadline exceeded"
```

**Pull request merged**